### DISCUSSION TITLE
Netherlands: ANWB Energie

### DISCUSSION THREAD

**Discussion by ikke-zelf** [JaccoR/hass-entso-e#79] [Price Modifyer Templates]
I used this one below for the ANWB Energie (from nov 20th when they started having their own).
The extra cost they charge is called 'Inkoopkosten'.
For now I believe they do not charge the 'cvo'

For the VAT I use the integrated option of this addon.
(note: do not forget to change from 9% to 21% at the 1st of jan.)

```
{% set s = {"extra_cost": 0.01472,"energie_belasting": 0.04010,"ode_belasting": 0.03325 } %} {{(current_price + s.extra_cost + s.energie_belasting + s.ode_belasting) | float}}
```

**Comment by jimsliep** [comment]
where are these cost refer to ? (  {"extra_cost": 0.01472,  )   I was able to find the other current tax cost on https://www.anwb.nl/huis/energie/energiebelasting

**Reply by ikke-zelf** [reply]
inkoopkosten

**Reply by jimsliep** [reply]
Was ik even vergeten, dankjewel

**Reply by jimsliep** [reply]
Waar kan ik die inkoopkosten terug vinden bij anwb, voor het geval die een keer verranderen?

**Reply by Houdoe** [reply]
When I look in the 'energiekostentabel' on the anbw energy  website (after login)  I can read the different components of how the energy costs are calculated: 
Date (hour)                  Fri, jan 6, 19:00 €0.15 ex vat
Verbruik (kWh)                              5.67
Leveringskosten                            0.83 (5.67 * 0.15=0.85)
Energiebelasting                           0.70 (5.67 * 0.1260=0.71)
Inkoopkosten                                0.10 
Netbeheerkosten                          0.03
ODE                                               0.00
Vaste leveringskosten                    0.01
Vermindering Energiebelasting    -0.06
BTW                                              0.33977
Totaal                                           1.95770 / 5.67 = 0.3452 / kwh this hour

Inkoopkosten: 0.10/5.67 = € 0.0176 / kWh
Netbeheerkosten enexis: € 325.90 (tax included): € 269.33 ex vat: per day 0.74 -> 0.03 per hour
Vaste leveringskosten: € 73.00 (tax included): € 60.33 ex vat: per day 0.17 -> 0.007 per hour rounded to 0.01?

My calculator does give a different calculation of 'BTW' and 'Totaal'. When and how are the component rounded in the price?

(sorry had ook in gewoon nederlands kunnen worden verteld)
Heb ook bij ANWB om nadere toelichting gevraagd. (ben benieuwd)

gebruik nu:

{% set s = {"extra_cost": 0.01472,"energie_belasting": 0.1524,"ode_belasting": 0.0 } %} {{(current_price + s.extra_cost + s.energie_belasting + s.ode_belasting) | float}}

And how is the field 'extra costs' calculated? now 0.01472 (tax included)

Ik lees overigens op de site van enever.nl/energieleveranciers dat de toeslag € 0.02118 incl BTW zou zijn.

Er lijkt sprake te zijn van 2 verschillende methoden:
1. zie boven (ANWB energiekostentabel) de prijs die opgebouwd is uit onderdelen vast kosten per dag en verbrukskosten op basis van aantal verbruikte kWh lees wat je daadwerkelijk moet gaan betalen.
2. een prijsberekening per kWh waarbij geen rekening is gehouden met de vaste elementen. (volgens mij wordt deze getoond in hone assistant.

**Reply by JeroenGeubbels** [reply]
Ik heb je formule toegepast, echter zie ik een groot verschil in de prijs die ik in HomeAssistant zie, en de prijs die de ANWB zelf aangeeft. Klopt dit wel?
![image](https://user-images.githubusercontent.com/47889023/211263928-4b0905bf-02e3-4d25-a8d7-29eb8df8a857.png)
 
![image](https://user-images.githubusercontent.com/47889023/211263973-5948f6aa-275f-42b8-870e-174de6bd54a0.png)

**Reply by Damazinho** [reply]
Omdat ANWB kale tarieven laat zien en in je grafiek zijn alle kosten verwerkt.

**Reply by JeroenGeubbels** [reply]
@Damazinho oke, goed te weten! Jammer dat ze doen lijken alsof het heel goedkoop is, en er nog % wat kosten bij komen.

**Reply by Houdoe** [reply]
> Omdat ANWB kale tarieven laat zien en in je grafiek zijn alle kosten verwerkt.

Volgens mij zitten de vaste kosten bestanddelen (kosten per dag) zie na inlog energie anwb website de energiekostentabel, er niet in.
Die laten zich lastig berekenen per verbruikt kWh.
Dit is nog veel extremer bij de kosten van het gas. Bij mij 5 dagen in januari 2.25 m3 kosten: € 6.55 dan is dus € 2.91 per m3. Maar dat komt dus door de vaste kosten per dag. De opgave van de ANWB gasprijs per m3 bij enever.nl is voor vandaag € 1.4907

kosten per dag zijn: "Vaste leveringskosten", "Netbeheerkosten" en "Vermindering Enegiebelasting"

**Comment by groetg** [comment]
Dus de meest actuele (vandaag 5 juni 2023) formule voor stroom is:
{% set s = {"extra_cost": 0.01472,"energie_belasting": 0.1524,"ode_belasting": 0.0 } %} {{(current_price + s.extra_cost + s.energie_belasting + s.ode_belasting) | float}}
Is dat correct?

**Comment by MarcHagen** [comment]
Can someone also explain the email we just got from ANWB ?

> Energiebelasting en inkoopkosten
> De stroom zelf is dan wel gratis, maar ook dan betaal je energiebelasting en inkoopkosten. > Er komt per kWh altijd nog 17 cent bij.

Where every template here I only get 0.08 cents per kWh, they talk about 17 cents? Do we miss something in our templates ?

**Reply by eelco-teuthof** [reply]
I'm using this formula 

{% set s = {"extra_cost": 0.0175,"energie_belasting": 0.12599,"VAT": 1.21} %} {{((current_price + s.extra_cost + s.energie_belasting) * s.VAT) | float}}

**Reply by rjvbeek** [reply]
I've updated this for 2024 to:

`{% set s = {"extra_cost": 0.04,"energie_belasting": 0.1084297520661157,"VAT": 1.21} %} {{((current_price + s.extra_cost + s.energie_belasting) * s.VAT) | float}}`

**Reply by MarcHagen** [reply]
Thank you! Works flawlessly, gives the correct output.
This website is really awesome in explaining how and giving a check on our template outputs: 
https://doe-duurzaam.nl/dossier/dynamische-energietarieven/

**Reply by jmlk52** [reply]
Excellent, thanks! To be precise de inkoopkosten (ANWB extra charges) 0,0484 (source ANWB website) and energie belasting you mentioned is exclusive BTW which is correct for the formulae. I wrongly used the figure mentioned by ANWB but that's including BTW so my calculated price per Kwh was too high. You helped me with the correct figure for Energie Belasting.

**Reply by jmlk52** [reply]
Apologies, I made again same mistake (with or without VAT (BTW). ANWB inkoopkosten should be 0,04 (ex VAT), which means MarcHagen is correct! I took the price from the ANWB website, but that inclusieve VAT.

**Reply by ckruijntjens** [reply]
What formule are you guys using for gas?

**Reply by RabbITCybErSeC** [reply]
Yes, very keen on knowing what you use for gas. I am using the energy zero integration for ANWB right now. But, this trying to figure you the formulea for that one. Wow this is a rabbit-hole

**Reply by rjvbeek** [reply]
I am using the following:

```
template:
  - sensor:
      - name: "Gasprijs"
        state: >
          {{ ((states('sensor.energyzero_today_gas_current_hour_price') | float + 0.0488512396694215 + 0.5838842975206612) * 1.21) | float }}
        unit_of_measurement: EUR/m³
```

**Reply by RabbITCybErSeC** [reply]
@rjvbeek how do you get to the numbers in the formula? Especially the:  `0.0488512396694215`, `0.583884297520661`

**Reply by rjvbeek** [reply]
I checked ANWB website for costs:
![image](https://github.com/JaccoR/hass-entso-e/assets/11409533/99e092cd-c3ca-48dd-8871-f8ca98ffcc3e)
And tax:
![image](https://github.com/JaccoR/hass-entso-e/assets/11409533/1edb2384-f500-4000-81ad-048b2e6a3080)
Then I subtracted VAT from those, and then I multiply everything by 1.21 after adding it up.

**Reply by RabbITCybErSeC** [reply]
@rjvbeek Thanks that makes sense now. Time to put the formula in HASS!

**Reply by ckruijntjens** [reply]
I Used your formula. Is it correct that the all incl costs for 1kwh with dynamic is now @19:12

0,41103 €/kWh

**Reply by rjvbeek** [reply]
41 cents seems a bit high. This is my data now:
![Screenshot_20240113_100833_Home Assistant](https://github.com/JaccoR/hass-entso-e/assets/11409533/5b745511-6622-48e8-9121-7da48bf2eaf3)

**Reply by ckruijntjens** [reply]
Are you also using this template?

{% set s = {"extra_cost": 0.04,"energie_belasting": 0.1084297520661157,"VAT": 1.21} %} {{((current_price + s.extra_cost + s.energie_belasting) * s.VAT) | float}}

my statics looking like this:

![image](https://github.com/JaccoR/hass-entso-e/assets/6084672/1c71d8ef-7e27-4323-9c30-f72852c86d5a)

**Reply by ckruijntjens** [reply]
![image](https://github.com/JaccoR/hass-entso-e/assets/6084672/4b3ebd94-7411-4e13-83f6-42221a78abee)

**Comment by ckruijntjens** [comment]
What formule do you use for solar panels. Deliver back to the net?

**Comment by Palerm0** [comment]
I've just lookt up the Belasting per KwH now (02-10-2024) on the website of ANWB. it shows 2024 "€0,1317"
ODE* (Opslag Duurzame Energie- en Klimaattransitie) is 0,0 (merged with the belasting as of 1-1-2023)

For 2025 the tax (Belasting) seems to be unchanged and wil stay @"€0,1317" 

This makes the formula:
{% set s = {"extra_cost": 0,04,"energie_belasting": 0,1317,"VAT": 1.21} %} {{((current_price + s.extra_cost + s.energie_belasting) * s.VAT) | float}}

Is this correct and the same as you guys have configured now?

Greetings
Palermo

[Source](https://www.anwb.nl/energie/energiebelasting)

**Reply by nschermer** [reply]
As the site states, the 0.0484 and 0.1228 are already including VAT, so the correct formula should be:

{% set s = {"extra_cost": 0.0484,"energie_belasting": 0.1228,"VAT": 1.21} %} {{(current_price * s.VAT + s.energie_belasting + s.extra_cost) | float}}

**Reply by akamming** [reply]
I used this formula, but pricing still seems to be a bit high. I noticed in this Formula there is a VAT. But in the integration i can also configure a VAT. So i think it is calculated twice.

Is my assumption correct that if i use the formula "{% set s = {"extra_cost": 0.0484,"energie_belasting": 0.1228,"VAT": 1.21} %} {{(current_price * s.VAT + s.energie_belasting + s.extra_cost) | float}}" that then the VAT in the integration must be set to 0 te get the correct pricing?

To be specific: This is my config now for ANWB: Is this correct? 
![image](https://github.com/user-attachments/assets/36d0ea8c-4098-4309-b4b2-aaa2745b77bf)

**Reply by E-m-r-y-s** [reply]
You are running the calculation using VAT in the formula. The VAT is added by default and should not be part of the Price Modifier Template.

Based on todays status with ANWB Energy the formula should be:
{% set s = {
  "inkoop_kosten": 0.01652,
  "energie_belasting": 0.10154,
  "ode_belasting": 0.0 
  } 
  %} 
  {{   (current_price + s.inkoop_kosten + s.energie_belasting + s.ode_belasting) | float}}


inkoopkosten are 0.02 including VAT, giving 0.01652 Ex VAT
energie belasting = as defined on the webstite from [belastingdienst](https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/overige_belastingen/belastingen_op_milieugrondslag/energiebelasting/)  0.10154 Ex VAT
ODO belasting is included in the energie belasting and could be dropped if needed.



The APEX-Chart setup would be in line of:
Be noted, I am using a conditional setup to hide the "Tomorrow Prices" if a certain time has not been reached (13:00 in my case)

the fully copy the APEX Chart Setup as below, first create a helper:
-> Times of the Day Sensor
name: Dynamic Prices Tomorrow Available
after : 01:00 PM
before 00:00 AM



type: vertical-stack
cards:
  - type: custom:apexcharts-card
    graph_span: 24h
    hours_12: false
    experimental:
      color_threshold: true
    header:
      title: Energie Prijs (Vandaag)
      show: true
      show_states: true
    span:
      start: day
    now:
      show: true
      label: nu
      color: "#ff9800"
    series:
      - entity: sensor.average_electricity_price
        type: column
        extend_to: end
        unit: €/kWh
        float_precision: 4
        yaxis_id: Price
        show:
          in_header: before_now
          extremas: true
          header_color_threshold: true
        color_threshold:
          - value: 0.2
            color: 00ed01
          - value: 0.22
            color: 3af901
          - value: 0.24
            color: 87fa00
          - value: 0.26
            color: cefb02
          - value: 0.28
            color: eeff00
          - value: 0.3
            color: ffde1a
          - value: 0.32
            color: ffa700
          - value: 0.34
            color: ff8d00
          - value: 0.36
            color: ff7400
          - value: 0.38
            color: ff4d00
          - value: 0.4
            color: ff4d00
          - value: 0.42
            color: ff0000
          - value: 0.44
            color: e60000
          - value: 0.46
            color: cc0000
          - value: 0.48
            color: b30000
          - value: 0.5
            color: "990000"
          - value: 0.52
            color: "800000"
          - value: 0.55
            color: "660000"
          - value: 0.6
            color: 4d0000
          - value: 0.75
            color: "330000"
        data_generator: |
          return entity.attributes.prices_today.map((entry) => {
            return [new Date(entry.time), entry.price];
          });
        group_by:
          func: avg
          duration: 1h
    yaxis:
      - id: Price
        min: 0
        max: 0.5
        decimals: 2
        apex_config:
          title:
            text: €/kWh
          tickAmount: 10
    apex_config:
      legend:
        show: false
      tooltip:
        x:
          show: true
          format: HH:00 - HH:59
  - type: conditional
    conditions:
      - condition: state
        entity: binary_sensor.dynamic_prices_tomorrow_available
        state: "on"
    card:
      type: custom:apexcharts-card
      graph_span: 24h
      hours_12: false
      experimental:
        color_threshold: true
      header:
        title: Energie Prijs (Morgen)
        show: true
        show_states: false
      span:
        start: day
        offset: +1day
      series:
        - entity: sensor.average_electricity_price
          type: column
          extend_to: end
          unit: €/kWh
          float_precision: 3
          yaxis_id: Price
          show:
            in_header: before_now
            extremas: true
            header_color_threshold: true
          color_threshold:
            - value: 0.2
              color: 00ed01
            - value: 0.22
              color: 3af901
            - value: 0.24
              color: 87fa00
            - value: 0.26
              color: cefb02
            - value: 0.28
              color: eeff00
            - value: 0.3
              color: ffde1a
            - value: 0.32
              color: ffa700
            - value: 0.34
              color: ff8d00
            - value: 0.36
              color: ff7400
            - value: 0.38
              color: ff4d00
            - value: 0.4
              color: ff4d00
            - value: 0.42
              color: ff0000
            - value: 0.44
              color: e60000
            - value: 0.46
              color: cc0000
            - value: 0.48
              color: b30000
            - value: 0.5
              color: "990000"
            - value: 0.52
              color: "800000"
            - value: 0.55
              color: "660000"
            - value: 0.6
              color: 4d0000
            - value: 0.75
              color: "330000"
          data_generator: |
            return entity.attributes.prices_tomorrow.map((entry) => {
              return [new Date(entry.time), entry.price];
            });
          group_by:
            func: avg
            duration: 1h
      yaxis:
        - id: Price
          min: 0
          max: 0.5
          decimals: 2
          apex_config:
            title:
              text: €/kWh
            tickAmount: 10
      apex_config:
        legend:
          show: false
        tooltip:
          x:
            show: true
            format: HH:00 - HH:59
title: Stroom Prijzen per uur

**Comment by justusmeijsen** [comment]
De formule voor 2026 van ANWB Energie is als volgt: 

`{% set s = {
"extra_cost": 0.018,
"energie_belasting_2026": 0.11085,
"VAT": 1.21
} %}{{((current_price) * s.VAT + s.extra_cost + s.energie_belasting_2026) | float}}`

**Comment by E-m-r-y-s** [comment]
Probably not for today as ENSTO-E is kinda down for NL prices..
Instead of doing things all manually and in chart, it might be better to look into the [EPEX Spot Sensor](https://github.com/mampfes/ha_epex_spot) 

You can setup the ENTSO-E data via this integration and by the integration after setup click on the wheel for settings). update the pricing options as below
<img width="1146" height="1205" alt="image" src="https://github.com/user-attachments/assets/32111922-7a3f-4e24-bc87-f88e25fa390e" />

All you need now is using the Total Price Sensor as provided via EPEX Spot and be done with it.. no more manual updating charts every little price change, but just 1 small update in the sensor setting and done

**Reply by pyxis1** [reply]
How did you calculate this 0,106486? E.g. how to know what to put there in 2027

**Reply by E-m-r-y-s** [reply]
Easy:
Energiebelasting for 2026 per kWh = € 0.09161 ex VAT
source: https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/overige_belastingen/belastingen_op_milieugrondslag/energiebelasting/energiebelasting

Inkoop charge ANWB ENergy per kWh = €0,018 INCL VAT
source: https://www.anwb.nl/energie/actuele-tarieven
So that makes it 0,018 / 1,21 = 0,01487603305785123966942148760331

The two numbers summed = € 0,106486

**Reply by pyxis1** [reply]
Thanks.
